bug: AVL strip-forces 500 on aircraft with decimal-named NACA airfoils (regression of gh-409)

## Observed Behavior

Running **Strip-Force Analysis** (Trefftz Plane tab) on an aircraft whose wing references a NACA airfoil with a **decimal** in its name (e.g. `naca23013.5`) returns HTTP 500:

```
Strip forces failed (500): Strip forces analysis error: AVL didn't produce stability output.
Check avl_command and input geometry.
stderr/stdout hint:
===================================================
Athena Vortex Lattice
Program Version 3.40
===================================================
Reading file: airplane.avl ...
Configuration: RV-7
Building surface: rudder
```

The truncated stdout (`stderr_text[:500]` in `app/services/avl_runner.py:352`) misleadingly ends on the rudder. The **full** AVL stdout — captured by running the AVL binary directly on the same generated `.avl` — reveals the actual crash:

```
   Building surface: rudder
   Building surface: elevator
   Building duplicate image-surface: elevator (YDUP)
   Building surface: main_wing
 ** Read error on line   77 ...
 23013.5
```

Line 77 of the generated `.avl` is:

```
SECTION
!  Xle    Yle    Zle    Chord  Ainc   Nspan  Sspace
   0.1921  0.0  -0.0445  0.183  0.0
NACA
23013.5         ← line 77, AVL parse error
CLAF
   1.1039796923873082
```

AVL's `NACA` keyword expects an **integer** 4- or 5-digit code (e.g. `2412`, `23012`). `23013.5` is not a valid NACA designation — the digit groups of the NACA series are integer-valued by definition. AVL fails parsing.

## Expected Behavior

When the airfoil reference contains a decimal (or any other character that makes it an invalid NACA designation), the `.avl` generator should emit `AFIL <path-to-dat-file>` instead of `NACA <digits>`. The `.dat` file already exists at `components/airfoils/naca23013.5.dat` and is used (correctly) for **other sections** of the same wing — only the first section incorrectly takes the NACA branch.

## Root Cause

`app/services/avl_geometry_service.py:40`:

```python
_NACA_RE = _re.compile(r"^naca\s*(\d{4,5}(?:\.\d+)?)$", _re.IGNORECASE)
```

The regex **explicitly allows** a decimal-fraction tail (`(?:\.\d+)?`). When `xsec.airfoil = "naca23013.5"` the regex matches and the code emits `AvlNaca(digits="23013.5")` → serialized to `NACA\n23013.5\n` in the `.avl`. Other sections of the same wing reference the airfoil by full path (`/.../naca23013.5.dat`), the regex doesn't match, the code falls through to file resolution → `AvlAfile(...)` → `AFIL /.../naca23013.5.dat` in the `.avl`. AVL accepts the `AFIL` form; it rejects the `NACA` form.

### Introducing commit

`86ed1233 fix(gh-409): resolve 500 error on strip_forces with decimal NACA airfoils (#410)` — 2026-05-05.

Ironic: that PR attempted to fix the very class of bug we now hit. It relaxed the regex to **accept** decimals at the wrong layer; the correct fix is to **reject** decimals so the airfoil ref falls through to file resolution (the working path used by all other sections). The PR also added two unit tests (`test_naca_decimal_thickness`, `test_naca_decimal_thickness_case_insensitive`) that pin the **wrong** contract (`isinstance(result, AvlNaca)` for `"naca23013.5"`). The fix here must invert those tests.

### Empirical verification

Patching the offending `.avl` in place — replacing the two lines `NACA\n23013.5` with `AFIL\n/Users/.../naca23013.5.dat` — and re-running the AVL binary yields:

```
   Building surface: rudder
   Building surface: elevator
   Building duplicate image-surface: elevator (YDUP)
   Building surface: main_wing
```

No `Read error`. Confirms the fix.

## Severity

**High.** Any aircraft with a non-standard NACA airfoil name (decimal thickness, custom variant) cannot run Strip-Force Analysis. The RV-7 in the local default dataset is affected. Pre-existing since 2026-05-05; only surfaced today because the user exercised the Trefftz Plane tab on a real aircraft after #577 made it more discoverable.

## Steps to Reproduce

1. Aircraft with a wing section whose `xsec.airfoil` string is `naca<digits>.<decimal>` (e.g. `naca23013.5`) — the local "RV-7" template is affected.
2. Open the Trefftz Plane tab, select any trimmed OP (or use manual mode), Run.
3. Backend returns 500. `analyze_airplane_strip_forces` → `AVLRunner.run` → `FileNotFoundError("AVL didn't produce stability output...")`.

## Environment

- **Backend version:** `087f4bb8` (post-gh-577).
- **AVL:** version 3.40 (vendored via `avl-binary` Python wheel).
- **Aircraft tested:** local "RV-7" (UUID `de75f07d-...`).

## Proposed Fix

1. `app/services/avl_geometry_service.py:40` — tighten the regex to integer-only:
   ```python
   _NACA_RE = _re.compile(r"^naca\s*(\d{4,5})$", _re.IGNORECASE)
   ```
2. `app/tests/test_avl_generator_integration.py:191-208` — invert `test_naca_decimal_thickness` and `test_naca_decimal_thickness_case_insensitive` to assert `isinstance(result, AvlAfile)` (file-resolution path), not `AvlNaca`.
3. Add a new integration-style test that asserts the generated `.avl` for a wing with a decimal-named airfoil contains `AFIL <path>` and **not** `NACA <decimal>`.
4. Mention the same gh-409 NeuralFoil-CDCL fix is still needed (and works correctly) — only the avl_geometry_service half was wrong.

## Additional Context

- The 500-char truncation in `avl_runner.py:352` (`stderr_text[:500]`) is itself a usability bug — it cut off the actually-useful `** Read error on line 77 ...` line. Worth widening to e.g. 4 KB and including the **tail** of the output, not the head, since AVL prints diagnostic output near the end. Out of scope for this ticket; tracked separately if needed.
- The `.dat` file at `components/airfoils/naca23013.5.dat` is generated upstream (presumably by an `Airfoil(name="naca23013.5")` round-trip via aerosandbox). That generation works; only the AVL serialization is broken.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## 🏷️ has-reproduction

> Label `has-reproduction` added by **supercycle/bug** · 2026-05-20

---

### Reproduction tests

**File:** `app/tests/test_avl_generator_integration.py`

Two existing gh-409 tests inverted (they pinned the wrong contract), one new integration assertion added:

- `TestBuildAirfoilNode.test_naca_decimal_thickness_falls_through_to_file` — asserts `_build_airfoil_node("naca23013.5")` returns `AvlAfile` with `.filepath.endswith("naca23013.5.dat")`.
- `TestBuildAirfoilNode.test_naca_decimal_thickness_case_insensitive_falls_through_to_file` — same with `"NACA23013.5"`.
- `TestBuildAirfoilNode.test_generated_avl_uses_afil_not_naca_for_decimal_airfoil` — asserts the serialised `.avl` content contains `AFIL` + `naca23013.5.dat` and does **not** contain `NACA\n23013.5`.

### Failing output (RED)

```
FAILED test_naca_decimal_thickness_falls_through_to_file
  AssertionError: assert isinstance(result, AvlAfile) — got AvlNaca(digits='23013.5')
FAILED test_naca_decimal_thickness_case_insensitive_falls_through_to_file
  same
FAILED test_generated_avl_uses_afil_not_naca_for_decimal_airfoil
  AssertionError: assert 'AFIL' in 'NACA\n23013.5\n'
3 failed, 3 passed in 0.15s
```

### Out-of-band empirical reproduction

Direct AVL run on the user's RV-7 `.avl`:

```
   Building surface: rudder
   Building surface: elevator
   Building duplicate image-surface: elevator (YDUP)
   Building surface: main_wing
 ** Read error on line   77 ...
 23013.5
```

After patching the offending two lines (`NACA\n23013.5` → `AFIL\n.../naca23013.5.dat`), AVL reads all surfaces without error.

### Existing tests that no longer fit and were inverted

`gh-409` PR #410 added `test_naca_decimal_thickness` and `test_naca_decimal_thickness_case_insensitive`, each asserting `isinstance(result, AvlNaca)` for `"naca23013.5"` and `result.digits == "23013.5"`. That contract is what produces the broken `.avl`. Both tests are rewritten in this branch to assert the **correct** routing (`AvlAfile`).

--- a/app/services/avl_geometry_service.py
+++ b/app/services/avl_geometry_service.py
@@ -35,9 +35,21 @@ from app.schemas.avl_geometry import AvlGeometryResponse
 logger = logging.getLogger(__name__)
 
 # ---------------------------------------------------------------------------
-# Regex for detecting NACA 4/5-digit airfoil names
+# Regex for detecting valid NACA 4-/5-digit airfoil names.
+#
+# Must be **integer** digit groups only — the NACA series itself is defined
+# with integer-valued parameters (e.g. NACA 2412, NACA 23012). AVL's ``NACA``
+# keyword expects an integer code and raises ``Read error`` on any decimal
+# value. Decimal-bearing names such as ``naca23013.5`` are custom airfoils
+# that exist as ``.dat`` files under ``components/airfoils/`` — they fall
+# through to file resolution (``_resolve_airfoil_reference`` → ``AFIL``)
+# instead of being passed verbatim to AVL's NACA generator.
+#
+# gh-588: the previous pattern ``\d{4,5}(?:\.\d+)?`` (from gh-409) over-matched
+# and routed ``naca23013.5`` into ``AvlNaca``, crashing AVL with
+# ``Read error on line N`` during ``Building surface: main_wing``.
 # ---------------------------------------------------------------------------
-_NACA_RE = _re.compile(r"^naca\s*(\d{4,5}(?:\.\d+)?)$", _re.IGNORECASE)
+_NACA_RE = _re.compile(r"^naca\s*(\d{4,5})$", _re.IGNORECASE)
 
 
 # ---------------------------------------------------------------------------

--- a/app/tests/test_avl_generator_integration.py
+++ b/app/tests/test_avl_generator_integration.py
@@ -191,21 +191,49 @@ class TestBuildAirfoilNode:
         assert result.digits == "0012"
 
 
-    def test_naca_decimal_thickness(self):
-        from app.avl.geometry import AvlNaca
+    def test_naca_decimal_thickness_falls_through_to_file(self):
+        """gh-588 (regression of gh-409): a decimal in a NACA-style name is NOT
+        a valid 4-/5-digit NACA designation. AVL's ``NACA`` keyword expects an
+        integer code (e.g. 2412, 23012) and crashes with ``Read error`` on a
+        decimal value. The generator must therefore route ``naca23013.5`` to
+        the file-resolution branch (``AFIL <path/naca23013.5.dat>``), which
+        AVL accepts.
+        """
+        from app.avl.geometry import AvlAfile
         from app.services.avl_geometry_service import _build_airfoil_node
 
         result = _build_airfoil_node("naca23013.5")
-        assert isinstance(result, AvlNaca)
-        assert result.digits == "23013.5"
+        assert isinstance(result, AvlAfile), (
+            "Decimal-named airfoils must resolve to AFIL so AVL can parse the "
+            "geometry; emitting NACA with a decimal designation triggers a "
+            "Read error on the AVL side (gh-588)."
+        )
+        assert result.filepath.endswith("naca23013.5.dat")
 
-    def test_naca_decimal_thickness_case_insensitive(self):
-        from app.avl.geometry import AvlNaca
+    def test_naca_decimal_thickness_case_insensitive_falls_through_to_file(self):
+        """gh-588: same as above with upper-case spelling."""
+        from app.avl.geometry import AvlAfile
         from app.services.avl_geometry_service import _build_airfoil_node
 
         result = _build_airfoil_node("NACA23013.5")
-        assert isinstance(result, AvlNaca)
-        assert result.digits == "23013.5"
+        assert isinstance(result, AvlAfile)
+        assert result.filepath.endswith("naca23013.5.dat")
+
+    def test_generated_avl_uses_afil_not_naca_for_decimal_airfoil(self):
+        """gh-588 end-to-end: the serialised ``.avl`` content for a decimal-named
+        airfoil must contain ``AFIL <path>`` and must NOT contain a bare
+        ``NACA\\n23013.5`` line that AVL would reject.
+        """
+        from app.services.avl_geometry_service import _build_airfoil_node
+
+        node = _build_airfoil_node("naca23013.5")
+        serialised = repr(node)
+        assert "AFIL" in serialised
+        assert "naca23013.5.dat" in serialised
+        # The bare decimal designation must not appear on its own line — AVL's
+        # NACA-keyword parser would choke on it.
+        assert "NACA\n23013.5" not in serialised
+        assert "23013.5\n" != serialised  # not a stand-alone integer line
 
 
 class TestBuildGeometryEdgeCases:

